src/events/mod.rs

```diff
@@ -135,7 +135,7 @@ impl Event {
         // User PUT event's into the homeserver write new data. We fetch the data
         // for every Resource Type
         let url = reqwest::Url::parse(&self.uri)?;
-        let blob = match self.pubky_client.get(url).await {
+        let blob = match PubkyConnector::get_pubky_client()?.get(url).await {
```

Comment:
for redability, what if we do something like this
```suggestion
        let pubky_client = PubkyConnector::get_pubky_client()?;
        let blob = match pubky_client.get(url).await {
```

This same comment might apply to other places as well, I did not look for all of the instances.